### PR TITLE
react-images => react-bnb-gallery + code splitting

### PR DESCRIPTION
**Pull request by apepper**
`ThumbnailGallery` previously used [`react-images`](https://www.npmjs.com/package/react-images). Now it uses [`react-bnb-gallery`](https://www.npmjs.com/package/react-bnb-gallery).

According to [`react-images`](https://www.npmjs.com/package/react-images)'s own readme:

> ⚠️  Warning!
>
> Don't use this in a new project. This package hasn't been properly maintained in a long time and there are much better options available.

So we needed a replacement.

Thankfully @pietruszka found a good library, that is quite an easy replacement: [`react-bnb-gallery`](https://www.npmjs.com/package/react-bnb-gallery). It works very similar to `react-images` and with a few tweaks looks at least similar.

### File size

To address filesize concerns about `react-bnb-gallery` we introduces code-splitting to cut of this part from the rest of javascript. The gallery only needs to be loaded, after a click happend.

### Screenshots

*Previously*
react-images

*Now*
react-bnb-gallery v2

### Open Todos
* [x] Check if it works fine with latest react version (without warnings)
* [x] Check if it works fine with IE11.
* [x] Check bundle size (from 999kb to 954kb)
* [x] Fine tune design by Alex G.

### CODE REVIEW

**Comment by agessler** [comment]
@apepper From my site ok. I changed by CSS the font-family and the background color of the overlay. I don't like to use `!important` that much. I would like to prefer changing more tags or reduce inline CSS styles.
The Font issue results of an `<h3>` use in the template I could not access. Overlay has inline CSS color hard coded.

**Comment by apepper** [comment]
@agessler Thanks for your design improvements.

> I changed by CSS [...] the background color of the overlay. I don't like to use !important that much.

@agessler I switch it to the official `backgroundColor` option.

**Review by dcsaszar — CHANGES_REQUESTED** [review]
Looks nice!

See comments. Also:

* [ ] Minor issue:
  * Open the gallery before confirming the cookie banner
  * Try to confirm the cookie banner, in order to be able to navigate the photo list
  * Expected: the cookie banner disappears
  * Actual: nothing happens when I click the buttons



* [ ] FYI Some style issues that are already in the original demo of react-bnb-gallery, so I guess we can't do much about them:
  * On mobile, when navigating the items, the photo list jumps up/down depending on the image description length.
  * On wide screens the photo list is right-aligned, where centered (around the current selection) would feel more natural.

File: webpack.config.js [hunk 1]
```diff
@@ -79,6 +79,7 @@ function webpackConfig(env = {}) {
                     },
                   ],
                 ],
+                plugins: ["@babel/plugin-syntax-dynamic-import"],
```
[the remark this answers is not in the document]
Comment (apepper) [reply webpack.config.js:82]: Nice!

File: src/Widgets/ThumbnailGalleryWidget/ThumbnailGalleryWidgetComponent.js [hunk 2]
```diff
@@ -98,17 +79,13 @@ class ThumbnailGalleryComponent extends React.Component {
...
-            onClickThumbnail={this.gotoImage}
+          <ReactBnbGallery
+            show={this.state.lightboxIsOpen}
+            backgroundColor="rgba(22,22,22,.9)"
+            activePhotoIndex={this.state.currentImage}
+            photos={lightboxImages}
             onClose={this.closeLightbox}
-            showThumbnails
-            backdropClosesModal
+            wrap={false}
```
[the remark this answers is not in the document]
Comment (apepper) [reply src/Widgets/ThumbnailGalleryWidget/ThumbnailGalleryWidgetComponent.js:88]: There is an option issue about the `wrap` property: https://github.com/peterpalau/react-bnb-gallery/issues/42 . I wanted it to make it more explicit in the hopes, that a future version will switch the meaning of "wrap=true".

**Comment by apepper** [comment]
REVIEW (d68325e)

Regarding the cookie consent issue: It seems to me, that this is an issue of either `react-bnb-gallery` and `react-cookie-consent`. I don't see an easy way to resolve the issue.

But since this only happens, if one opens the dialog I think we can live with this for now.

File: src/Widgets/ThumbnailGalleryWidget/ThumbnailGalleryWidgetComponent.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dcsaszar) [reply src/Widgets/ThumbnailGalleryWidget/ThumbnailGalleryWidgetComponent.js:88]: The default behaviour is good. The rest is speculation and later versions may not match your hopes.

**Comment by dcsaszar** [comment]
> But since this only happens, if one opens the dialog I think we can live with this for now.

In general, I agree this is a minor thing and can be fixed later, but:

Imagine a first-time visitor. They can scroll around the homepage and click anywhere.
As far as I can see there are only 2 areas where they cannot do this and are required to close the cookie banner:
* The footer
* An opened gallery photo list

Seen this way, 50% of the use cases for closing the cookie banner are broken.

**Comment by dcsaszar** [comment]
As a quickfix:
If cookie box and gallery use the same `zIndex` it feels more natural, because then the gallery hides the cookiebox.

The gallery default appears to be 2000, for the cookie box we set 1000000.

**Comment by apepper** [comment]
REVIEW (31432a4)

Fixed cookie banner issue: The gallery now overlaps the cookie banner fully.

**Review by dcsaszar — APPROVED** [review]
👍

**Pull request merged**